cli/internal/cmd/configgenerate.go

```diff
@@ -135,13 +171,39 @@ func parseGenerateFlags(cmd *cobra.Command) (generateFlags, error) {
 		return generateFlags{}, fmt.Errorf("resolving kuberentes version from flag: %w", err)
 	}
 
+	attestationString, err := cmd.Flags().GetString("attestation")
+	if err != nil {
+		return generateFlags{}, fmt.Errorf("parsing attestation flag: %w", err)
+	}
+
+	var attestationType variant.Variant
+	// if no attestation type is specified, use the default for the cloud provider
+	if attestationString == "" {
+		attestationType = variant.Default{}
+	} else {
+		resType := validateAttestationType(attestationString)
```

Comment:
you can use `variant.FromString(attestationString)` to return a variant (and error) from a string